superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx

```diff
@@ -232,6 +262,20 @@ class DatasourceControl extends React.PureComponent {
       </Menu>
     );
 
+    const queryDatasourceMenu = (
+      <Menu onClick={this.handleMenuItemClick}>
+        <Menu.Item key={QUERY_PREVIEW}>{t('Query preview')}</Menu.Item>
+        <Menu.Item key={VIEW_IN_SQL_LAB}>{t('View in SQL Lab')}</Menu.Item>
+        <Menu.Item key={SAVE_AS_DATASET}>{t('Save as dataset')}</Menu.Item>
+      </Menu>
+    );
+
+    const datasourceTypeCheck =
```

Comment:
do we have this somewhere else? Also the current dataset is SlTable.. but I think I remember seeing us using Dataset?

Reply:
Oh whoops, good catch! This already exists as `isValidDatasourceType` in `DatasourcePanel/index.tsx` here: https://github.com/apache/superset/blob/master/superset-frontend/src/explore/components/DatasourcePanel/index.tsx#L310-L314

So I'll move/export/use that one instead.

But it does include SlTable, as well as Dataset. I think it was set up that way to check for any of the DatasourceTypes (minus table and druid). Should I remove it?

Reply:
Ok, since this still lives in two places, can we dry it up a bit with something like 
`Object.values(DatasourceType).includes(datasource.type)`

Reply:
nm, I see that you removed it in the other component. Still possibly a good way to make it more succinct.